tests/drivers/flash/stm32/testcase.yaml

```diff
@@ -54,3 +54,11 @@ tests:
       - CONFIG_FLASH_STM32_WRITE_PROTECT=y
     filter: dt_compat_enabled("st,stm32h7-flash-controller") and
       dt_label_with_parent_compat_enabled("storage_partition", "fixed-partitions")
+  drivers.flash.stm32.h7.block_registers:
+    platform_allow:
+      - nucleo_h743zi
+      - google_icetower
+    extra_configs:
+      - CONFIG_FLASH_STM32_BLOCK_REGISTERS=y
+    filter: dt_compat_enabled("st,stm32h7-flash-controller") and
+            dt_label_with_parent_compat_enabled("storage_partition", "fixed-partitions")
```

Comment:
This filter will also match for h7rs based platforms. Is that expected ?

Reply:
No, since it is not supported for h7rs. Isn't platform allow enough?

Reply:
Right, but then maybe filter on `CONFIG_SOC_SERIES_STM32H7XX` instead of this compatible which cover both series to be consistent.

Reply:
Done.